AISKU/Tests/TestFramework/TestClass.ts

```diff
@@ -221,6 +240,32 @@ class TestClass {
             JSON.stringify(data));
     }
 
+    public getPayloadMessages(spy:SinonSpy, ignoreInit:boolean = false) {
+        let resultPayload = [];
+        if (spy.called && spy.args && spy.args.length > 0) {
+            spy.args.forEach(call => {
+                call[0].forEach(message => {
+                    // Ignore the internal SendBrowserInfoOnUserInit message (Only occurs when running tests in a browser)
+                    if (ignoreInit || message.indexOf("AI (Internal): 72 ") === -1) {
+                        resultPayload.push(message);
+                    }
+                })
+            });
+        }
+
+        return resultPayload;
+    }
+
+    public dumpPayloadMessages(spy:SinonSpy) {
+        let msg = "Sent Messages";
+        if (spy.called && spy.args && spy.args.length > 0) {
+            spy.args[0][0].forEach((value, idx) => {
+                msg += "\n" + idx + ":" + value;
+            });
+        }
+        Assert.ok(false, msg);
```

Comment:
Why is `msg` false, isn't it always has a value?

Reply:
So that the message is always logged into the output, as this is only called when the tests fail.